text/0517-io-os-reform.md

```diff
@@ -1370,6 +1370,91 @@ The contents of `std::io::net` submodules `tcp`, `udp`, `ip` and
 the other modules are being moved or removed and are described
 elsewhere.
 
+#### InetAddr
+
+The composition of an `IpAddr` and a port. It has the following interface:
+
+```rust
+impl InetAddr {
+  /// Returns a new InetAddr composed of an unspecified v4 IP and a 0
+  /// port
+  fn any_v4() -> InetAddr;
+
+  /// Returns a new InetAddr composed of an unspecified v6 IP and a 0
+  /// port
+  fn any_v6() -> InetAddr;
+
+  fn ip(&self) -> IpAddr;
+  fn port(&self) -> u16;
+
+  /// Returns true if the IpAddr is unspecified and port == 0
+  fn is_unspecified(&self) -> bool;
+}
+```
+
+#### IpAddr
+
+Represents an IP address. It has the following interface:
+
+```rust
+impl IpAddr {
+  fn new_v4(a: u8, b: u8, c: u8, d: u8) -> IpAddr;
```

Comment:
Is there a reason there isn't a new that accepts an `Ipv4Addr`? Or is there a conversion provided that I'm overlooking?